src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -718,14 +713,16 @@ private static ExpressionSyntax ParseIntegerJToken(JValue value)
             return SyntaxFactory.CreatePositiveOrNegativeInteger(value.Value<long>());
         }
 
-        private SyntaxBase ParseJValue(JValue value)
+        private SyntaxBase ParseJValue(JValue value, bool permitExpressions)
             => value.Type switch
             {
-                JTokenType.String => ParseString(value.ToString(CultureInfo.InvariantCulture), value),
-                JTokenType.Uri => ParseString(value.ToString(CultureInfo.InvariantCulture), value),
+                JTokenType.String or 
+                JTokenType.Uri or 
+                JTokenType.Date or
+                JTokenType.Float => permitExpressions ? 
+                    ParseString(value.ToString(CultureInfo.InvariantCulture), value) : 
+                    SyntaxFactory.CreateStringLiteral(value.ToString(CultureInfo.InvariantCulture)),
```

Comment:
Does `JTokenType.Float` belong here? I thought that was decompiled to `json('<float>')`

Reply:
Good catch. This is the existing behavior, but I don't see any harm in fixing it as part of this PR.